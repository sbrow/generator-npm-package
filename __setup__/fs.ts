import { readFileSync } from "fs";
import { join } from "path";

/**
 * Returns the contents of file with name `file` from the directory
 * at path `dir` as a parsed object.
 *
 * @export
 * @param {string} dir
 * @param {string} file
 * @returns {*}
 */
export function loadJSON(dir: string, file: string): any {
  return JSON.parse(load(dir, file));
}

export function load(dir: string, file: string): string {
  return readFileSync(join(dir, file)).toString();
}
