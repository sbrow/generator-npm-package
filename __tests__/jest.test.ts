import assert from "yeoman-assert";
import fs from "fs";
import helpers from "yeoman-test";
import path from "path";
import Jest from "../src/jest/index";

const opts = {
    resolved: require.resolve("../src/jest/index"),
    namespace: "npm-package:jest",
};
describe("generator-jest", () => {
    describe("#writing", () => {
        const file = "jest.config.js";
        it("creates file", async () => {
            const tmpDir = await helpers.run(Jest, opts);
            assert.file(file);
        });
        it(`extensions = [".js"]`, async () => {
            const tmpDir = await helpers.run(Jest, opts);
            const got = fs.readFileSync(path.join(tmpDir, file)).toString();
            expect(got).toMatchSnapshot();
        });
        it(`extensions = [".js", ".jsx"]`, async () => {
            const tmpDir = await helpers.run(Jest, opts)
                .withLocalConfig({ devDependencies: ["react"] });
            const got = fs.readFileSync(path.join(tmpDir, file)).toString();
            expect(got).toMatchSnapshot();
        });
        it(`extensions = [".ts", ".js"]`, async () => {
            const tmpDir = await helpers.run(Jest, opts)
                .withLocalConfig({
                    devDependencies: ["typescript"],
                    tsconfig: { resolveJsonModule: false },
                })
            const got = fs.readFileSync(path.join(tmpDir, file)).toString();
            expect(got).toMatchSnapshot();
        });
        it(`extensions = [".ts", ".js", ".json"]`, async () => {
            const tmpDir = await helpers.run(Jest, opts)
                .withLocalConfig({
                    devDependencies: ["typescript"],
                    tsconfig: { resolveJsonModule: true },
                })
            const got = fs.readFileSync(path.join(tmpDir, file)).toString();
            expect(got).toMatchSnapshot();
        });
        it(`extensions = [".ts", ".tsx", ".js", ".jsx"]`, async () => {
            const tmpDir = await helpers.run(Jest, opts)
                .withLocalConfig({
                    devDependencies: ["react", "typescript"],
                    tsconfig: { resolveJsonModule: false },
                })
            const got = fs.readFileSync(path.join(tmpDir, file)).toString();
            expect(got).toMatchSnapshot();
        });
    });
});
