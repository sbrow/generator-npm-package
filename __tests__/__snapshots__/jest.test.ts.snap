// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator-jest #writing extensions = [".js", ".jsx"] 1`] = `
"const moduleFileExtensions = [\\"js\\",\\"jsx\\"]

module.exports = {
    collectCoverageFrom: [\`src/**/*.\${moduleFileExtensions}\`]
}
"
`;

exports[`generator-jest #writing extensions = [".js"] 1`] = `
"const moduleFileExtensions = [\\"js\\"]

module.exports = {
    collectCoverageFrom: [\`src/**/*.\${moduleFileExtensions}\`]
}
"
`;

exports[`generator-jest #writing extensions = [".ts", ".js", ".json"] 1`] = `
"const moduleFileExtensions = [\\"ts\\",\\"js\\",\\"json\\"]

module.exports = {
    collectCoverageFrom: [\`src/**/*.\${moduleFileExtensions}\`]
}
"
`;

exports[`generator-jest #writing extensions = [".ts", ".js"] 1`] = `
"const moduleFileExtensions = [\\"ts\\",\\"js\\"]

module.exports = {
    collectCoverageFrom: [\`src/**/*.\${moduleFileExtensions}\`]
}
"
`;

exports[`generator-jest #writing extensions = [".ts", ".tsx", ".js", ".jsx"] 1`] = `
"const moduleFileExtensions = [\\"ts\\",\\"tsx\\",\\"js\\",\\"jsx\\"]

module.exports = {
    collectCoverageFrom: [\`src/**/*.\${moduleFileExtensions}\`]
}
"
`;
